core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -343,7 +345,8 @@ class Partition(val topic: String,
       leaderEpoch = partitionStateInfo.basePartitionState.leaderEpoch
       zkVersion = partitionStateInfo.basePartitionState.zkVersion
 
-      if (leaderReplicaIdOpt.isDefined && leaderReplicaIdOpt.get == newLeaderBrokerId) {
+      // If the leader is unchanged and the epochs are no more than one change apart, indicate that no follower changes are required
+      if (leaderReplicaIdOpt.isDefined && leaderReplicaIdOpt.get == newLeaderBrokerId && (leaderEpoch == oldLeaderEpoch || leaderEpoch == oldLeaderEpoch + 1)) {
```

Comment:
nit: I think we can replace `leaderReplicaIdOpt.isDefined && leaderReplicaIdOpt.get == newLeaderBrokerId` with `leaderReplicaIdOpt.exists(newLeaderBrokerId)`.